test/EtherFiAdminUpgrade.t.sol

```diff
@@ -0,0 +1,214 @@
+        _pubKey[12] = hex"a6c36df930f45b4c48057ae329519b6fa0d2a5ca78aaaa14781e2b5f36489c7c4385160a38b536ccb7c89278b11f38ef";
+        _pubKey[13] = hex"b8a9afe415e6facbc61c826a756e40c356343ccff993e95214699af3990d1df2effb984d7b7bfeb88a8df8403f5e846a";
+        _pubKey[14] = hex"a83049af5ac2c32be4134654177cd62d9c66f4aa4197499c6b5575946a4e91566649a224b5bf0fe0be5375767daa66cd";
+        _pubKey[15] = hex"b7ba13ed7b0590407b855e49bd3cb26b75959d140521a83750f7c5b71f93f4414eb47ebd625ee9ca56a5c097d550233e";
+        _pubKey[16] = hex"b293da30d0f4940dcf688dfa43bf7616b35c77269e7a08deaf91571a2c5ba604ca312f106fab51b538b114f59c6c9ca8";
+        _pubKey[17] = hex"918e73d0545f35c27c03c7da70cf26b1ffad545febeb0d9f2260c2b984389dd1c37c2c2a2481f62aae98bde55dcf3321";
+        _pubKey[18] = hex"86b7f539e7313987d67788cd7ed67d469ab98172dbab133144ba0bf8833be40747513a99e0b2f6530e521d4de114767b";
+        _pubKey[19] = hex"b758e8123dd22d2b534707d627a41d5e20a4eeef49b5a764f3770cbe4e3e66d16940350ce94f6d7c98e95ecbecdb156f";
+        _pubKey[20] = hex"9248a998630095a2373ea124dc358b5bfa7143373b22f043683f393a9ee553e1c9d6632d7da5be1f6699484087652875";
+        _pubKey[21] = hex"854c069f2437184ef0170aa6fd607489f3229f010d4fc647fc419ea4b5d062d2857d745bc3d9e1365e046c28e4099d65";
+        _pubKey[22] = hex"920f978f7aa95bae09ec450409083af9b6e8e8817a84b71763e72e568d0ff0e247b61eb01b60f32eb823969cfe720103";
+        _pubKey[23] = hex"a98772501ec0405f42e0b5b2b166b65b850c7c39a861946c5b6dc77c2caa5a34ac4fc8af8d2fdeac8991819e3db7fa93";
+        _pubKey[24] = hex"a5570902fea39478fbbbbac62862214d86fb87a7f51e5b690ddb97f230d89613100f2817894324cb09a09d259044f0ed";
+        return _pubKey;
+    }
+
+    function test_queueValidatorApproval() public {
+        uint256[] memory batchValidatorsToApprove = _getValidatorToApprove(52835, batchSize); //52835 based on transaction
+        bytes32 approvalHash = keccak256(abi.encode(reportHash, batchValidatorsToApprove));
+
+        //submit, execute report and approve validators
+        vm.startPrank(0x12582A27E5e19492b4FcD194a60F8f5e1aa31B0F);
+        etherFiOracleInstance.submitReport(report);
+        skip(3600);
+        etherFiAdminInstance.executeTasks(report);
+        (bool preCompleted, bool preExists) = etherFiAdminInstance.ValidatorApprovalTaskStatus(approvalHash);
+        assertTrue(preExists);
+        assertFalse(preCompleted);
+        etherFiAdminInstance.AdminTask_ApproveValidatorTask(reportHash, batchValidatorsToApprove, pubKeys, signatures);
+        (bool postCompleted, bool postExists) = etherFiAdminInstance.ValidatorApprovalTaskStatus(approvalHash);
+        assertTrue(postCompleted);
+        assertTrue(postExists);
+        vm.stopPrank();
+    }
+
+    //test invalidate
+    function test_invalidateReport() public {
+        uint256[] memory batchValidatorsToApprove = _getValidatorToApprove(52835, batchSize); //52835 based on transaction
```

Comment:
let's remove some dup code

for example, you can make the below test and call from other test functions too

```
function test_executeTasks() {
     uint256[] memory batchValidatorsToApprove = _getValidatorToApprove(52835, batchSize); //52835 based on transaction
        bytes32 approvalHash = keccak256(abi.encode(reportHash, batchValidatorsToApprove));

        //submit, execute report and approve validators
        vm.startPrank(0x12582A27E5e19492b4FcD194a60F8f5e1aa31B0F);
        etherFiOracleInstance.submitReport(report);
        skip(3600);
        etherFiAdminInstance.executeTasks(report);
}
```